knative-operator/pkg/monitoring/sources/source_deployment_discovery_controller.go

```diff
@@ -2,23 +2,31 @@ package sources
 
 import (
 	"context"
+	"os"
+	"strconv"
 
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/monitoring"
+	okomon "github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/monitoring"
 	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = common.Log.WithName("source-deployment-discovery-controller")
+var (
+	generateSourceServiceMonitorsEnvVar = "GENERATE_SERVICE_MONITORS_FOR_SOURCES_BY_DEFAULT"
+	useClusterMonitoringEnvVar          = "USE_CLUSTER_MONITORING_FOR_SOURCES_BY_DEFAULT"
```

Comment:
Similar comment as above. If not needed, ommit the `BY_DEFAULT` suffix.

Reply:
Another recommendation. If you start the env var name with "SOURCES" then its much clearer that those env vars belong together. Also, I wonder whether you do need two env vars.

E.g. you could differentiate on the value

```
SOURCES_SERVICE_MONITOR=cluster
SOURCES_SERVICE_MONITOR=generate
```

That makes the configurations surface smaller and less to remember.

Reply:
These are different settings. They affect different functionality pls take a look at the description.